### ISSUE TITLE
Plugin misses use of deprecated interface properties

### ISSUE THREAD

**Issue by ecraig12345** [gund/eslint-plugin-deprecation#5]
One feature this plugin is missing compared to the [tslint `deprecation` rule](https://palantir.github.io/tslint/rules/deprecation/) is detecting usage of deprecated interface/type properties. For example:

```ts
interface Thing1 {
  /** @deprecated */
  foo?: string;
}
const a: Thing1 = { foo: 'b' }; // should be an error
a.foo = 'c'; // correctly detected as error

type Thing2 = {
  /** @deprecated */
  foo?: string;
}
const b: Thing2 = { foo: 'c' }; // should be an error
b.foo = 'd'; // correctly detected as error
```

**Comment by gund** [comment]
Hi, as already mentioned - I cannot fix/improve this plugin at this point in time.

You are welcome to open a PR for this feature if you want.
